apis/rds/v1alpha1/zz_db_cluster.go

```diff
@@ -0,0 +1,475 @@
+	// The name of the database engine to be used for this DB cluster.
+	//
+	// Valid Values: aurora (for MySQL 5.6-compatible Aurora), aurora-mysql (for
+	// MySQL 5.7-compatible Aurora), and aurora-postgresql
+	// +kubebuilder:validation:Required
+	Engine *string `json:"engine"`
+	// The DB engine mode of the DB cluster, either provisioned, serverless, parallelquery,
+	// global, or multimaster.
+	//
+	// global engine mode only applies for global database clusters created with
+	// Aurora MySQL version 5.6.10a. For higher Aurora MySQL versions, the clusters
+	// in a global database use provisioned engine mode.
+	//
+	// Limitations and requirements apply to some DB engine modes. For more information,
+	// see the following sections in the Amazon Aurora User Guide:
+	//
+	//    * Limitations of Aurora Serverless (https://docs.aws.amazon.com/AmazonRDS/latest/AuroraUserGuide/aurora-serverless.html#aurora-serverless.limitations)
+	//
+	//    * Limitations of Parallel Query (https://docs.aws.amazon.com/AmazonRDS/latest/AuroraUserGuide/aurora-mysql-parallel-query.html#aurora-mysql-parallel-query-limitations)
+	//
+	//    * Requirements for Aurora Global Databases (https://docs.aws.amazon.com/AmazonRDS/latest/AuroraUserGuide/aurora-global-database.html#aurora-global-database.limitations)
+	//
+	//    * Limitations of Multi-Master Clusters (https://docs.aws.amazon.com/AmazonRDS/latest/AuroraUserGuide/aurora-multi-master.html#aurora-multi-master-limitations)
+	EngineMode *string `json:"engineMode,omitempty"`
+	// The version number of the database engine to use.
+	//
+	// To list all of the available engine versions for aurora (for MySQL 5.6-compatible
+	// Aurora), use the following command:
+	//
+	// aws rds describe-db-engine-versions --engine aurora --query "DBEngineVersions[].EngineVersion"
+	//
+	// To list all of the available engine versions for aurora-mysql (for MySQL
+	// 5.7-compatible Aurora), use the following command:
+	//
+	// aws rds describe-db-engine-versions --engine aurora-mysql --query "DBEngineVersions[].EngineVersion"
+	//
+	// To list all of the available engine versions for aurora-postgresql, use the
+	// following command:
+	//
+	// aws rds describe-db-engine-versions --engine aurora-postgresql --query "DBEngineVersions[].EngineVersion"
+	//
+	// Aurora MySQL
+	//
+	// Example: 5.6.10a, 5.6.mysql_aurora.1.19.2, 5.7.12, 5.7.mysql_aurora.2.04.5
+	//
+	// Aurora PostgreSQL
+	//
+	// Example: 9.6.3, 10.7
+	EngineVersion *string `json:"engineVersion,omitempty"`
+	// The global cluster ID of an Aurora cluster that becomes the primary cluster
+	// in the new global database cluster.
+	GlobalClusterIdentifier *string `json:"globalClusterIdentifier,omitempty"`
+	// The password for the master database user. This password can contain any
+	// printable ASCII character except "/", """, or "@".
+	//
+	// Constraints: Must contain from 8 to 41 characters.
+	MasterUserPassword *string `json:"masterUserPassword,omitempty"`
```

Comment:
We need to ignore this field and have a `Secret` reference to populate it. See the [example](https://github.com/crossplane/provider-aws/blob/087f587/apis/database/v1beta1/rdsinstance_types.go#L446) from `RDSInstance`.